src/test/java/org/apache/sling/models/impl/AdapterCacheHolderTest.java

```diff
@@ -0,0 +1,117 @@
+
+@RunWith(MockitoJUnitRunner.class)
+public class AdapterCacheHolderTest {
+
+    @Test
+    public void testGetMapNonServletRequestAdaptable() {
+        Map<Object, Map<Class<?>, SoftReference<Object>>> outerMap = spy(new HashMap<>());
+        Object adaptable1 = new Object();
+        Object adaptable2 = new Object();
+
+        try (AdapterCacheHolder adapterCacheHolder = new AdapterCacheHolder(false, outerMap)) {
+            Map<Class<?>, SoftReference<Object>> map1 = adapterCacheHolder.getCacheMapForAdaptable(adaptable1);
+            Map<Class<?>, SoftReference<Object>> map2 = adapterCacheHolder.getCacheMapForAdaptable(adaptable1);
+            Map<Class<?>, SoftReference<Object>> map3 = adapterCacheHolder.getCacheMapForAdaptable(adaptable2);
+
+            assertSame(map1, map2);
+            assertNotSame(map1, map3);
+        }
+    }
+
+    @Test
+    public void testGetMapServletRequestAdaptable() {
+        Map<Object, Map<Class<?>, SoftReference<Object>>> outerMap = spy(new HashMap<>());
+        SlingHttpServletRequest adaptable1 = new MockSlingHttpServletRequest(null);
+        SlingHttpServletRequest adaptable2 = new SlingHttpServletRequestWrapper(adaptable1);
+        SlingHttpServletRequest adaptable3 = new MockSlingHttpServletRequest(null);
+
+        try (AdapterCacheHolder adapterCacheHolder = new AdapterCacheHolder(false, outerMap)) {
+            Map<Class<?>, SoftReference<Object>> map1 = adapterCacheHolder.getCacheMapForAdaptable(adaptable1);
+            Map<Class<?>, SoftReference<Object>> map2 = adapterCacheHolder.getCacheMapForAdaptable(adaptable2);
+            Map<Class<?>, SoftReference<Object>> map3 = adapterCacheHolder.getCacheMapForAdaptable(adaptable3);
+
+            assertSame(map1, map2);
+            assertSame(map1, map3);
+        }
+    }
+
+    @Test
+    public void testClose() {
+        Object adaptable = new Object();
+        Object result = new Object();
+        SoftReference<Object> ref = new SoftReference<>(result);
+
+        Map<Class<?>, SoftReference<Object>> innerMap = spy(new HashMap<>());
+        innerMap.put(Object.class, ref);
+
+        Map<Object, Map<Class<?>, SoftReference<Object>>> outerMap = spy(new HashMap<>());
+        outerMap.put(adaptable, innerMap);
+
+        try (AdapterCacheHolder adapterCacheHolder = new AdapterCacheHolder(false, outerMap)) {
+            // intentionally empty
+        }
+
+        verify(outerMap).values();
+        verify(outerMap).clear();
+        verify(innerMap).clear();
+    }
+
+    @Test
+    public void testNewMapSyncMap() {
+        // This is a very weak test, but it should be sufficient to at least confirm that we are not just using a
+        //  WeakHashMap, in case someone changes it without knowing why.
+        Map<Object, Object> nonSyncMap= AdapterCacheHolder.newMap(false);
+        Map<Object, Object> syncMap= AdapterCacheHolder.newMap(true);
+
+        System.out.println(nonSyncMap.getClass());
```

Comment:
is this required here?

Reply:
Indeed they are not. Updated in latest push.